src/aggregate/aggregate_request.c

```diff
@@ -1461,7 +1460,8 @@ static void buildImplicitPipeline(AREQ *req, QueryError *Status) {
 
   RLookup_Init(first, cache);
 
-  ResultProcessor *rp = RPIndexIterator_New(req->rootiter);
+  ResultProcessor *rp = RPQueryIterator_New(req->rootiter);
```

Comment:
Should we have a constructor from from `Request`to hide this ownership transfer?

Reply:
IMO no, this API makes sense. With Rust, it will also be very clear